src/API.jl

```diff
@@ -904,25 +905,34 @@ function _is_stale(paths::Vector{String}, sourcepath::String, toml_c::Base.TOMLC
     return true
 end
 
+precompile_auto(b::Union{Bool,String}) = precompile_auto(Context(), b)
+precompile_auto(ctx::Context, s::String) = precompile_auto(ctx, parse(Bool, s))
+function precompile_auto(ctx::Context, b::Bool) 
+    ENV["JULIA_PKG_PRECOMPILE_AUTO"] = b ? "1" : "0"
+    nothing
+end
+
 function _auto_precompile()
-    if parse(Int, get(ENV, "JULIA_PKG_PRECOMPILE_AUTO", "0")) == 1
+    if parse(Int, get(ENV, "JULIA_PKG_PRECOMPILE_AUTO", "1")) == 1
         Pkg.precompile(internal_call=true)
     end
 end
 
-precompile(;internal_call::Bool=false) = precompile(Context(), internal_call=internal_call)
-function precompile(ctx::Context; internal_call::Bool=false)    
+precompile(; kwargs...) = precompile(Context(); kwargs...)
+function precompile(ctx::Context; internal_call::Bool=false, io::IO=stdout)    
     num_tasks = parse(Int, get(ENV, "JULIA_NUM_PRECOMPILE_TASKS", string(Sys.CPU_THREADS + 1)))
     parallel_limiter = Base.Semaphore(num_tasks)
-    
+    fancy_print = (io isa Base.TTY) && (get(ENV, "CI", nothing) != "true")
+
     # when manually called, unsuspend all packages that were suspended due to precomp errors
     !internal_call && Operations.precomp_unsuspend!() 
+    action_help = internal_call ? " (tip: control auto with `pkg> precompile_auto true/false`)" : ""
```

Comment:
Imo, it would be better to only tell about the env variable. With this, you would have to do it for every session and that seems annoying enough that you would want to learn how to do it the persistent way.

Reply:
Updated
![image](https://user-images.githubusercontent.com/1694067/95669302-69c77a00-0b4d-11eb-9f91-f89a832486f7.png)